Framework/Core/src/ResourcesMonitoringManager.h

```diff
@@ -0,0 +1,78 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef __RESOURCES_MONITORING_MANAGER__H__
+#define __RESOURCES_MONITORING_MANAGER__H__
+
+
+
+#include <unordered_map>
+#include <vector>
+#include <variant>
+#include <chrono>
+#include "Framework/DeviceMetricsInfo.h"
+#include <array>
+#include "Monitoring/ProcessMonitor.h"
+
+namespace o2
+{
+namespace framework
+{
+
+
+using PerformanceMetric_t = std::variant<double, uint64_t>;
+
+struct PerformanceMetric
+{
+    std::string m_timestamp;
+    std::string m_name;
+    std::string m_value;
+};
+
+class ResourcesMonitoringManager
+{
+public:
+
+    void enableResourcesMonitoring(unsigned short interval) noexcept
+        { m_enabled = true; m_interval = interval; }
+    void disableResourcesMonitoring() noexcept
+        { m_enabled = false; }
+
+    void addNewMetrics(const std::string& specName, std::vector<PerformanceMetric>&& parsedMetrics) noexcept;
+    bool dumpMetricsToJSON() noexcept;
```

Comment:
Why do you need yet another metrics store? Can't you simply use the DeviceMetricInfos one?